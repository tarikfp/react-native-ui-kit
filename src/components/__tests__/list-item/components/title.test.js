import * as React from "react";
import renderer from "react-test-renderer";
import ListItemTitle from "../../../list-item/components/title";

describe("tests list item title component", () => {
  it("renders list item title", () => {
    const tree = renderer.create(<ListItemTitle title="Test" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders list item title with subtitle", () => {
    const tree = renderer
      .create(<ListItemTitle title="Test" subtitle="Test subtitle" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders list item title with custom title style", () => {
    const tree = renderer
      .create(<ListItemTitle title="Test" style={{ fontSize: 40 }} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders list item title with custom subtitle style", () => {
    const tree = renderer
      .create(
        <ListItemTitle
          title="Test"
          subtitleStyle={{ fontSize: 40, fontWeight: "700" }}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders list item title with custom wrapper style", () => {
    const tree = renderer
      .create(
        <ListItemTitle title="Test" wrapperStyle={{ height: 70, padding: 8 }} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
