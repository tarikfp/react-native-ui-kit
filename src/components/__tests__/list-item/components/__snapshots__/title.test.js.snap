// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests list item title component renders list item title 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(35, 47, 72, 1)",
          "fontSize": 16,
          "fontWeight": "500",
          "lineHeight": 24,
          "textAlign": "left",
        },
        undefined,
      ]
    }
  >
    Test
  </Text>
</View>
`;

exports[`tests list item title component renders list item title with custom subtitle style 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(35, 47, 72, 1)",
          "fontSize": 16,
          "fontWeight": "500",
          "lineHeight": 24,
          "textAlign": "left",
        },
        undefined,
      ]
    }
  >
    Test
  </Text>
</View>
`;

exports[`tests list item title component renders list item title with custom title style 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(35, 47, 72, 1)",
          "fontSize": 16,
          "fontWeight": "500",
          "lineHeight": 24,
          "textAlign": "left",
        },
        Object {
          "fontSize": 40,
        },
      ]
    }
  >
    Test
  </Text>
</View>
`;

exports[`tests list item title component renders list item title with custom wrapper style 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "justifyContent": "center",
      },
      Object {
        "height": 70,
        "padding": 8,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(35, 47, 72, 1)",
          "fontSize": 16,
          "fontWeight": "500",
          "lineHeight": 24,
          "textAlign": "left",
        },
        undefined,
      ]
    }
  >
    Test
  </Text>
</View>
`;

exports[`tests list item title component renders list item title with subtitle 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(35, 47, 72, 1)",
          "fontSize": 16,
          "fontWeight": "500",
          "lineHeight": 24,
          "textAlign": "left",
        },
        undefined,
      ]
    }
  >
    Test
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(35, 47, 72, 1)",
          "fontSize": 14,
          "fontWeight": "500",
          "lineHeight": 20,
          "textAlign": "left",
        },
        undefined,
      ]
    }
  >
    Test subtitle
  </Text>
</View>
`;
